### PR TITLE
refactor(client/wounding): general cleanup

### PR DESCRIPTION
**Pull request by Manason**
**Describe Pull request**
- extracted functions
- introduced guard clauses
- simplified expressions

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? [no] (Be honest)
- Does your code fit the style guidelines? [yes]
- Does your PR fit the contribution guidelines? [yes]

### CODE REVIEW

File: client/wounding.lua [hunk 1]
```diff
@@ -176,84 +180,99 @@ CreateThread(function()
...
-                        bleedTickTimer = bleedTickTimer + 1
-                    end
-                end
-                bleedTickTimer = bleedTickTimer + 1
-            end
+local function advanceBleedTimer(player)
+    if math.floor(bleedTickTimer % (Config.BleedTickRate / 10)) == 0 then
+        local currPos = GetEntityCoords(player, true)
+        local moving = #(vector2(prevPos.x, prevPos.y) - vector2(currPos.x, currPos.y))
+        if (moving > 1 and not IsPedInAnyVehicle(player)) and isBleeding > 2 then
```
Comment (BerkieBb) [inline client/wounding.lua]: use cache.vehicle

File: client/wounding.lua [hunk 2]
```diff
@@ -176,84 +180,99 @@ CreateThread(function()
...
-                    else
-                        bleedTickTimer = bleedTickTimer + 1
-                    end
-                end
-                bleedTickTimer = bleedTickTimer + 1
-            end
+local function advanceBleedTimer(player)
+    if math.floor(bleedTickTimer % (Config.BleedTickRate / 10)) == 0 then
+        local currPos = GetEntityCoords(player, true)
+        local moving = #(vector2(prevPos.x, prevPos.y) - vector2(currPos.x, currPos.y))
```
Comment (BerkieBb) [inline client/wounding.lua]:
```suggestion
        local moving = #(prevPos.xy - currPos.xy)
```

File: client/wounding.lua [hunk 3]
```diff
@@ -176,84 +180,99 @@ CreateThread(function()
...
+        else
+            bleedTickTimer += 1
         end
     end
+    bleedTickTimer += 1
+end
+
+local function checkBleeding()
+    if isBleeding <= 0 or onPainKillers then return end
+    local player = PlayerPedId()
```
Comment (BerkieBb) [inline client/wounding.lua]: Use cache.ped

File: client/wounding.lua [hunk 4]
```diff
@@ -176,84 +180,99 @@ CreateThread(function()
...
+        handleBleeding(player)
+        bleedTickTimer = 0
+    else
+        advanceBleedTimer(player)
+    end
+end
+
+CreateThread(function()
+    Wait(2500)
+    prevPos = GetEntityCoords(PlayerPedId(), true)
```
Comment (BerkieBb) [inline client/wounding.lua]: Use cache.ped

File: client/main.lua [hunk 5]
```diff
@@ -15,7 +15,7 @@ inBedDict = "anim@gangops@morgue@table@"
 inBedAnim = "body_search"
 isInHospitalBed = false
 isBleeding = 0
-bleedTickTimer, advanceBleedTimer = 0, 0
+bleedTickTimer, advancedBleedTimer = 0, 0
```
Comment (BerkieBb) [inline client/main.lua]: This was actually meant to be `advance` because once it's over it advances to the next stage of bleeding/death

**Review by BerkieBb — CHANGES_REQUESTED** [review]

File: client/wounding.lua (hunk 1 above)
Comment (Manason) [reply client/wounding.lua]: done

**Review by BerkieBb — APPROVED** [review]

File: client/wounding.lua [hunk 6]
```diff
@@ -133,127 +129,150 @@ RegisterNetEvent('hospital:client:UsePainkillers', function()
...
 end)
 
--- Threads
+local function consumePainKiller()
+    painkillerAmount -= 1
+    Wait(Config.PainkillerInterval * 1000)
+    if painkillerAmount <= 0 then
+        painkillerAmount = 0
+        onPainKillers = false
+    end
```
Comment (YvanCywan) [inline client/wounding.lua]:
```suggestion
    if painkillerAmount > 0 then return end
    painkillerAmount = 0
    onPainKillers = false
```

**Review by YvanCywan — APPROVED** [review]
LGTM

**Pull request merged**